packages/core/useTransition/index.ts

```diff
@@ -136,25 +163,32 @@ function toVec(t: number | number[] | undefined) {
  * @param to
  * @param options
  */
-export function executeTransition<T extends number | number[]>(
+export function transition<T>(
   source: Ref<T>,
   from: MaybeRefOrGetter<T>,
   to: MaybeRefOrGetter<T>,
-  options: TransitionOptions = {},
+  options: TransitionOptions<T> = {},
 ): PromiseLike<void> {
   const {
     window = defaultWindow,
   } = options
   const fromVal = toValue(from)
   const toVal = toValue(to)
-  const v1 = toVec(fromVal)
-  const v2 = toVec(toVal)
   const duration = toValue(options.duration) ?? 1000
   const startedAt = Date.now()
   const endAt = Date.now() + duration
-  const trans = typeof options.transition === 'function'
-    ? options.transition
-    : (toValue(options.transition) ?? linear)
+
+  const interpolation = typeof options.interpolation === 'function'
+    ? options.interpolation
+    : defaultInterpolation
+
+  const trans = typeof options.easing !== 'undefined'
+    ? typeof options.easing === 'function'
+      ? options.easing
+      : (toValue(options.easing) ?? linear)
+    : typeof options.transition === 'function'
+      ? options.transition
+      : (toValue(options.transition) ?? linear)
```

Comment:
I don't think we need all of those checks.

If `options.easing` and `options.transition` are `MaybeRef<Fn>`, we can do the following:

```ts
  const trans = toValue(options.easing) ?? toValue(options.transition) ?? linear
```

Reply:
@OrbisK Done, I think this is a bit easier to read and removes a branch of redundant logic. When `options.transition` is removed, we can move the logic back into the `transition` function to live alongside `interpolation`

b49631e04df41ff07e8ce1405d514e4088c8b094

Reply:
([b49631e](https://github.com/vueuse/vueuse/pull/5011/commits/b49631e04df41ff07e8ce1405d514e4088c8b094) does work for me)